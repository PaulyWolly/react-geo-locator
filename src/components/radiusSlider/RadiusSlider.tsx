import React, { FC, ReactElement, useEffect, useState } from 'react';
import { Slider } from 'antd';

import './radiusSlider.styles.scss';

interface Islider {
  updateSlider: Function
}
const RadiusSlider: FC<Islider> = ({ updateSlider }): ReactElement => {
  const [sliderValue, setSliderValue] = useState(25);
  const onSliderChange = (e: any) => {
    const radiusInMeters = e * 1000;
    setSliderValue(e);
    updateSlider(radiusInMeters);
  }
  // on mount update innitial slider value
  useEffect(() => {
    updateSlider(sliderValue * 1000)
    // eslint-disable-next-line
  }, []);

  const marks: any = {
    5: '5km',
    50: {
      style: {
        color: 'rgba(0,0,0,0.7)',
      },
      label: <strong>50Km</strong>,
    },
  };
  return (
    <div className="slider-wrapper">
      <h4> Pick a geo-fencing radius</h4>
      <Slider
        min={5}
        max={50}
        className="slider"
        onChange={onSliderChange}
        marks={marks}
        value={sliderValue}
      />
    </div>
  )
}

export default RadiusSlider;