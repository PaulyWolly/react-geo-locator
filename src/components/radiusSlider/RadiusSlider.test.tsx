import React from 'react';
import { render, cleanup } from '@testing-library/react';
import RadiusSlider from './RadiusSlider';

afterEach(cleanup);

const mockedUpdateSlider: Function = () => { }

test('Renders <RadiusSlider /> component', () => {
  const { asFragment } = render(<RadiusSlider updateSlider={mockedUpdateSlider} />);
  expect(asFragment()).toMatchSnapshot();
});