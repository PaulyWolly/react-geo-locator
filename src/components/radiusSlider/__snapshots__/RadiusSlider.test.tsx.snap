// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <RadiusSlider /> component 1`] = `
<DocumentFragment>
  <div
    class="slider-wrapper"
  >
    <h4>
       Pick a geo-fencing radius
    </h4>
    <div
      class="ant-slider ant-slider-with-marks slider"
    >
      <div
        class="ant-slider-rail"
      />
      <div
        class="ant-slider-track"
        style="left: 0%; width: 44.44444444444444%;"
      />
      <div
        class="ant-slider-step"
      >
        <span
          class="ant-slider-dot ant-slider-dot-active"
          style="left: 0%;"
        />
        <span
          class="ant-slider-dot"
          style="left: 100%;"
        />
      </div>
      <div
        aria-disabled="false"
        aria-valuemax="50"
        aria-valuemin="5"
        aria-valuenow="25"
        class="ant-slider-handle"
        role="slider"
        style="left: 44.44444444444444%; transform: translateX(-50%);"
        tabindex="0"
      />
      <div
        class="ant-slider-mark"
      >
        <span
          class="ant-slider-mark-text ant-slider-mark-text-active"
          style="transform: translateX(-50%); left: 0%;"
        >
          5km
        </span>
        <span
          class="ant-slider-mark-text"
          style="transform: translateX(-50%); left: 100%; color: rgba(0, 0, 0, 0.7);"
        >
          <strong>
            50Km
          </strong>
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
